src/shared/user-record-nss.c

```diff
@@ -6,6 +6,13 @@
 #include "strv.h"
 #include "user-record-nss.h"
 
+#define SET_IF(field, condition, value, fallback)  \
+        if (condition) {                      \
+                field = value;                \
+        } else {                              \
+                field = fallback;             \
+        }
```

Comment:
maybe use:

```
field = (condition) ? (value) : (fallback);
```

feels better, since `field` is only evaluated once, and stronger type checking is done as value and fallback get checked for compatibility against each other.